src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -326,13 +326,26 @@ private ValueTask<HttpConnection> GetOrReserveHttp11ConnectionAsync(Cancellation
 
             // See if we have an HTTP2 connection
             Http2Connection http2Connection = _http2Connection;
+
+            if (http2Connection != null)
+            {
+                TimeSpan pooledConnectionLifetime = _poolManager.Settings._pooledConnectionLifetime;
+                // Check only lifetime expiration. There is no need to check idle if we are about to use it.
+                if (http2Connection.LifetimeExpired(DateTimeOffset.UtcNow, pooledConnectionLifetime))
+                {
+                    http2Connection.Dispose();
+                    InvalidateHttp2Connection(http2Connection);
```

Comment:
Don't we need to set http2Connection = null here? As it stands it looks like we will use the connection anyway. If you refactor the null check then maybe this doesn't matter.